WooCommerce/Classes/Authentication/Application Password/ApplicationPasswordAuthorizationWebViewController.swift

```diff
@@ -0,0 +1,222 @@
+                self?.handleAuthorizationResponse(with: url)
+            }
+            .store(in: &subscriptions)
+    }
+
+    func extendContentUnderSafeAreas() {
+        webView.scrollView.clipsToBounds = false
+        view.backgroundColor = webView.underPageBackgroundColor
+    }
+
+    func configureProgressBar() {
+        view.addSubview(progressBar)
+        NSLayoutConstraint.activate([
+            view.leadingAnchor.constraint(equalTo: progressBar.leadingAnchor),
+            view.trailingAnchor.constraint(equalTo: progressBar.trailingAnchor),
+            view.safeTopAnchor.constraint(equalTo: progressBar.topAnchor)
+        ])
+    }
+
+    func configureActivityIndicator() {
+        view.addSubview(activityIndicator)
+        NSLayoutConstraint.activate([
+            view.centerXAnchor.constraint(equalTo: activityIndicator.centerXAnchor),
+            view.centerYAnchor.constraint(equalTo: activityIndicator.centerYAnchor)
+        ])
+    }
+
+    func fetchAuthorizationURL() {
+        Task { @MainActor in
+            activityIndicator.startAnimating()
+            do {
+                guard let url = try await viewModel.fetchAuthURL() else {
+                    DDLogError("⛔️ No authorization URL found for application passwords")
+                    return showErrorAlert(message: Localization.applicationPasswordDisabled)
+                }
+                loadAuthorizationPage(url: url)
+            } catch {
+                DDLogError("⛔️ Error fetching authorization URL for application passwords \(error)")
+                showErrorAlert(message: Localization.errorFetchingAuthURL, onRetry: { [weak self] in
+                    self?.fetchAuthorizationURL()
+                })
+            }
+            activityIndicator.stopAnimating()
+        }
+    }
+
+    func loadAuthorizationPage(url: URL) {
+        let parameters: [URLQueryItem] = [
+            URLQueryItem(name: "app_name", value: appName),
+            URLQueryItem(name: "app_id", value: appID),
+            URLQueryItem(name: "success_url", value: Constants.successURL)
+        ]
+        var components = URLComponents(url: url, resolvingAgainstBaseURL: true)
+        components?.queryItems = parameters
+        guard let urlWithQueries = components?.url else {
+            DDLogError("⛔️ Error building authorization URL request")
+            return
+        }
+        let request = URLRequest(url: urlWithQueries)
+        webView.load(request)
+    }
+
+    func handleAuthorizationResponse(with url: URL?) {
+        guard let url, url.absoluteString.hasPrefix(Constants.successURL) else {
+            return
+        }
+        let components = URLComponents(url: url, resolvingAgainstBaseURL: true)
+        guard let queryItems = components?.queryItems,
+              let username = queryItems.first(where: { $0.name == "user_login" })?.value,
+              let password = queryItems.first(where: { $0.name == "password" })?.value else {
```

Comment:
Nit: Consider moving the strings to `Constants`.

Reply:
Done in 772985bfa9ecff86de96c5cd21e6fe6d501769fa